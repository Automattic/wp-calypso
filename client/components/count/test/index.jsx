/**
 * External dependencies
 */
var expect = require( 'chai' ).expect,
	useMockery = require( 'test/helpers/use-mockery' );

describe( 'Count', function() {
	var React, ReactClass, TestUtils, Count, renderer;

	// really only using Mockery for the clean module cache
	useMockery();

	before( function() {
		React = require( 'react' );
		ReactClass = require( 'react/lib/ReactClass' );
		TestUtils = require( 'react-addons-test-utils' );

		ReactClass.injection.injectMixin( require( 'i18n-calypso' ).mixin );
		Count = require( '../' );
	} );

	beforeEach( function() {
		renderer = TestUtils.createRenderer();
	} );

	it( 'should render the passed count', function() {
		var result;

		renderer.render( <Count count={ 23 } /> );
		result = renderer.getRenderOutput();

		expect( result.props.className ).to.equal( 'count' );
		expect( result.props.children ).to.equal( '23' );
	} );

	it( 'should use the correct class name', function() {
		var result;

		renderer.render( <Count count={ 23 } /> );
		result = renderer.getRenderOutput();

		expect( result.props.className ).to.equal( 'count' );
	} );

	it( 'should internationalize the passed count', function() {
		var result;

		renderer.render( <Count count={ 2317 } /> );
		result = renderer.getRenderOutput();

		expect( result.props.children ).to.equal( '2,317' );
	} );

	it( 'should render zero', function() {
		var result;

		renderer.render( <Count count={ 0 } /> );
		result = renderer.getRenderOutput();

		expect( result.props.children ).to.equal( '0' );
	} );

	it( 'should render negative numbers', function() {
		var result;

		renderer.render( <Count count={ -1000 } /> );
		result = renderer.getRenderOutput();

		expect( result.props.children ).to.equal( '-1,000' );
	} );

	it( 'should cut off floating point numbers', function() {
		var result;

		renderer.render( <Count count={ 3.1415926 } /> );
		result = renderer.getRenderOutput();

		expect( result.props.children ).to.equal( '3' );
	} );
} );
