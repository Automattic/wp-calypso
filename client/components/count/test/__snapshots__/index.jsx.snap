// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Count should render with primary class 1`] = `
<span
  class="count is-primary"
>
  23
</span>
`;
